spring-boot-cli/src/main/java/org/springframework/boot/cli/command/CommandRunner.java

```diff
@@ -185,17 +185,29 @@ public int runAndHandleErrors(String... args) {
 	}
 
 	private String[] removeDebugFlags(String[] args) {
-		List<String> rtn = new ArrayList<String>(args.length);
+		List<String> argsWithoutDebugFlags = new ArrayList<String>(args.length);
 		boolean appArgsDetected = false;
 		for (String arg : args) {
 			// Allow apps to have a -d argument
-			appArgsDetected |= "--".equals(arg);
-			if (("-d".equals(arg) || "--debug".equals(arg)) && !appArgsDetected) {
+			appArgsDetected |= isAppArg(arg);
+			if (isDebugArgTogetherWithAppArg(arg, appArgsDetected)) {
 				continue;
 			}
-			rtn.add(arg);
+			argsWithoutDebugFlags.add(arg);
 		}
-		return rtn.toArray(new String[rtn.size()]);
+		return argsWithoutDebugFlags.toArray(new String[argsWithoutDebugFlags.size()]);
```

Comment:
I'll do another pull request from a new branch on my fork, since I messed up a little with git rebase and I'll include your patch